### PR TITLE
fix(core): incorrect CloudFormation date formatting

### LINKED ISSUE #28065: s3 LifecycleRule.ExpirationDate: due to changes in CFN validation last week - template format is broken
### Describe the bug

Since last week our template failed to deploy. After talking to AWS customer services seems that last week a strict validation was forced now on this field. which leads the CDK output template to fail.

today the output results this text 

`"ExpirationDate": "2024-11-19T00:00:00"`

as understood from the customer services - `Z` must be added in the end.

our code is doing so:
`lifecycleMaxSizeRuleBuilder.expirationDate(LocalDate.now().plus(1, ChronoUnit.YEARS).atStartOfDay().toInstant(ZoneOffset.UTC));`

### Expected Behavior

With Z in the end
`"ExpirationDate": "2024-11-19T00:00:00Z"`

### Current Behavior

`"ExpirationDate": "2024-11-19T00:00:00"`

### Reproduction Steps

simply synth

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.109.0

### Framework Version

_No response_

### Node.js Version

16.13.1

### OS

Win 10

### Language

Java

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by msambol**
The trailing `Z` is now required: 

- [ExpirationDate](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-rule.html#cfn-s3-bucket-rule-expirationdate)
- [TransitionDate](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-transition.html#cfn-s3-bucket-transition-transitiondate)

Closes #28065, #28124.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/cli-lib-alpha/THIRD_PARTY_LICENSES
Comment (lpizzinidev) [inline packages/@aws-cdk/cli-lib-alpha/THIRD_PARTY_LICENSES:1]: Should this file be included?

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Looks good 👍
I left you a note on an added file.

Also, the `dateToCloudFormation` method is used when parsing dates for [`ScheduledAction`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-applicationautoscaling-scalabletarget-scheduledaction.html)'s `StartTime` and `EndTime`.
Can you please add integration/unit tests for that as well?
(You can use [this](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk-testing/framework-integ/test/aws-lambda/test/integ.autoscaling.lit.ts) as a starting point).

Finally, I'm not sure about the title, it should [describe the bug (not the solution)](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-4-pull-request).
Something like `fix(core): wrong CloudFormation date formatting` is probably more aligned.

File: packages/@aws-cdk/cli-lib-alpha/THIRD_PARTY_LICENSES
Comment (wakeupmh) [reply packages/@aws-cdk/cli-lib-alpha/THIRD_PARTY_LICENSES:1]: I think he added the EOF only

**Review by wakeupmh — APPROVED** [review]

**Comment by msambol** [comment]
@lpizzinidev I updated the PR with the other integration test. 👍🏼

**Review by SankyRed — APPROVED** [review]
The change looks good to me. Thank you for your contribution.

**Pull request merged**